runtimebp/signal.go

```diff
@@ -17,6 +17,17 @@ var defaultSignals = []os.Signal{
 	syscall.SIGTERM,
 }
 
+// State is an enum representing the runtime state of the baseplate server
+type State int
+
+const (
+	//TODO docs
+	StateUnknown State = iota
+	StateStarting
+	StateServing
+	StateDraining
+)
```

Comment:
🔕 I'm not sure that it makes sense to have this in runtimebp while the rest of it is in baseplate? It should probably all just be in there IMO.